### DISCUSSION TITLE
adoption pitfalls

### DISCUSSION THREAD

**Discussion by seeden** [tamagui/tamagui#75] [General]
In this thread, I would like to summarise the pitfalls during my Tamagui adoption.
It is not critique. I just want to highlight my problems and help other people.

1. I expected that all styles would work out of the box. I spent half of the day looking at why it is not working with my development environment. The main reason was simple: injectCSS is disabled by default. I understand that SSR is an outstanding feature. I am using it as well. But for newbies, it is a first pain in the ass. 
A probably better idea is to enable it by default, and when somebody wants to use SSR, he will find it in the documentation on how to disable it.

3. Simple working example with React. I found https://github.com/tamagui/tamagui/tree/master/packages/sandbox, which was almost what I wanted to use, but it is not working when you start to use other standards react libraries. Other libraries have problems with custom createElement instead of standard React.createElement. When I tried to use standard React.createElement, tamagui had a problem with the non-existing React package. I needed to add one more plugin into the webpack config, which fixed React 18 build: (2 hours of my life)

```js
    new ProvidePlugin({
	 React: 'react',
   }),
```

4. Nonworking documentation/github links and inconsistent documentation. (I totally understand but it is tough these days)

5. I have no idea why I need to setup importsWhitelist :) I read the documentation multiple times, but I have no idea what I need to add to this variable. Do I need to add @tamagui/theme-base when I am using it? Why is tamagui.config.ts not there? This field is confusing for me.

6. I like @tamagui/theme-base. It is excellent. What is missing or the most wanted feature by me is to create a theme with my own "primary" color. Default colors are great for documentation and examples, but users prefer their own primary/secondary colors. Probably export auto generate function will be enough. I am not sure if "export function createThemes" with named radix color will be enough.

I will add more items in short runs

**Comment by natew** [comment]
@seeden I think adding a "Common Problems" or FAQ or something section to the top part of the docs and we can just keep adding new sections like this.

I do need to explain importsWhitelist better, I'll do that now.

Primary color would be great, I just added that myself to my up next.

**Reply by natew** [reply]
And of course generation is up next for themes.
